desktop/apps/auction2/actions.js

```diff
@@ -12,11 +42,112 @@ export function updateArtworks(artworks) {
   }
 }
 
-export function toggleListView(isListView) {
+export function updateAggregatedArtists(aggregatedArtists) {
   return {
-    type: TOGGLE_LIST_VIEW,
+    type: UPDATE_AGGREGATED_ARTISTS,
     payload: {
-      isListView
+      aggregatedArtists
+    }
+  }
+}
+
+export function updateAggregatedMediums(aggregatedMediums) {
+  return {
+    type: UPDATE_AGGREGATED_MEDIUMS,
+    payload: {
+      aggregatedMediums
+    }
+  }
+}
+
+export function updateArtistParams(artistId) {
+  return (dispatch) => {
+    dispatch(updateArtistId(artistId))
+    dispatch(fetchArtworks())
+  }
+}
+
+export function updateArtistId(artistId) {
+  return {
+    type: UPDATE_ARTIST_ID,
+    payload: {
+      artistId
     }
   }
 }
+
+export function updateMediumParams(mediumId) {
+  return (dispatch) => {
+    dispatch(updateMediumId(mediumId))
+    dispatch(fetchArtworks())
+  }
+}
+
+export function updateMediumId(mediumId) {
```

Comment:
These kinds of action creators are pretty slim. Do we want to explicitly define each one? Or have some kind of meta-action that accepts a `type` and `payload`. This felt more idiomatic but am open to suggestions!